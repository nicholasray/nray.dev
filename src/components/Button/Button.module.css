.button {
  @apply inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50;
}

.buttonDefault {
  @apply bg-primary text-primary-foreground shadow hover:bg-primary/90;
}

.buttonPrimary {
  @apply bg-primary text-primary-foreground shadow hover:bg-primary/90;
}

.buttonDestructive {
  @apply bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90;
}

.buttonOutline {
  @apply border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground;
}

.buttonSecondary {
  @apply bg-secondary text-secondary-foreground shadow-sm hover:bg-secondary/80;
}

.buttonGhost {
  @apply hover:bg-accent hover:text-accent-foreground;
}

.buttonLink {
  @apply text-primary underline-offset-4 hover:underline;
}

.sizeDefault {
  @apply h-9 px-4 py-2;
}

.sizeSm {
  @apply h-8 rounded-md px-3 text-xs;
}

.sizeLg {
  @apply h-8 rounded-md px-3 text-xs;
}

.sizeIcon {
  @apply h-9 w-9;
}
