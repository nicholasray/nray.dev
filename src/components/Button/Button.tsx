import * as React from "react";
import { Slot } from "@radix-ui/react-slot";
import { cva, type VariantProps } from "class-variance-authority";
import styles from "./Button.module.css";

const buttonVariants = cva(styles.button, {
  variants: {
    variant: {
      default: styles.buttonDefault,
      destructive: styles.buttonDestructive,
      outline: styles.buttonOutline,
      secondary: styles.buttonSecondary,
      ghost: styles.buttonGhost,
      link: styles.buttonLink,
    },
    size: {
      default: styles.sizeDefault,
      sm: styles.sizeSm,
      lg: styles.sizeLg,
      icon: styles.sizeIcon,
    },
  },
  defaultVariants: {
    variant: "default",
    size: "default",
  },
});

export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof buttonVariants> {
  asChild?: boolean;
}

const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
  ({ className, variant, size, asChild = false, ...props }, ref) => {
    const Comp = asChild ? Slot : "button";
    return (
      <Comp
        className={buttonVariants({ variant, size, className })}
        ref={ref}
        {...props}
      />
    );
  },
);
Button.displayName = "Button";

export { Button, buttonVariants };
