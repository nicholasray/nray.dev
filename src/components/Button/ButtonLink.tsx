import React from "react";
import { Button, type ButtonProps } from "./Button";

export interface LinkProps extends ButtonProps {
  href: string;
  children: React.ReactNode;
}

const ButtonLink: React.FC<LinkProps> = ({ href, children, ...props }) => {
  return (
    <Button {...props} asChild>
      <a href={href}>{children}</a>
    </Button>
  );
};

export { ButtonLink };
