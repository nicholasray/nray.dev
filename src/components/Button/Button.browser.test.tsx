import { Button } from "./Button";
import { render } from "vitest-browser-react";

describe("Button", () => {
  it("renders correctly", async () => {
    const screen = render(<Button>Submit</Button>);

    await expect.element(screen.getByText("Submit")).toBeVisible();

    expect(screen.container.firstChild).toMatchSnapshot();
  });
});
