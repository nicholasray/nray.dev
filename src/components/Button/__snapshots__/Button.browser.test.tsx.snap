// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly 1`] = `
<button
  class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
>
  Submit
</button>
`;
